docs/api.rst

```diff
@@ -22,16 +22,24 @@ The :class:`NoisyChannels` class
 
    NoisyChannels
 
-The :class:`PrepPipeline` class
--------------------------------
+The :class:`~pyprep.Reference` class
+------------------------------------
+
+.. autosummary::
+   :toctree: generated/
+
+   Reference
+   
+The :class:`~pyprep.PrepPipeline` class
+---------------------------------------
```

Comment:
I am not sure whether or not this should be made public (including the import in __init__.py) --> for `NoisyChannels`, I can see why people may want to use it separately from `PrepPipeline` --> but do such usecases also exist for a barebones `Reference`?

A general rule is to expose as little API as possible ... and only as much as necessary :) That helps tremendously with development (backward incompatible changes) and also helps users to get a clear picture of what can be done and the reasoning behind it (not everybody is a power user).

WDYT?

Reply:
Fully understood about wanting to keep the public API minimal. My _personal_ use-case for `Reference` being used separately from `PrepPipreline` (apart from my MATLAB comparison efforts where I've pre-cleanlined the data) is that I want to be able to plot the results of Cleanline separately from the results of full PREP (cleanline + re-reference) in order to sanity-check the process at each step for each file (see this chunk of my script [here](https://github.com/LBRF/DEMI_Analysis_Pipeline/blob/8534580133d260f7adc0259b4d85c4e2e4ea7d3f/_Preprocessing/eeg_pipeline.py#L259-L296)).

Apart from that use-case, the other main one I can think of would be if someone wanted to use a different line noise filtering method but still wanted to use PyPREP's robust re-referencing. That said, I guess you could get all the same functionality by restructuring `PrepPipeline` a bit to allow for separate CleanLine and/or re-referencing from that class instead of making `Reference` public. If that sounds better to you (or if you'd just prefer to keep the public API as-is), I'll just remove `Reference` from the init along with my references to it in any public docstrings.

Reply:
okay cool, that sounds like a fair use. +1 to making it part of the public API then, as proposed here. In the future we should then add an example of how to make use of that (like you do in your script).